lib/engine/share_pool.rb

```diff
@@ -207,10 +207,16 @@ def change_president(presidents_share, swap_to, president)
 
       num_shares = presidents_share.percent / corporation.share_percent
 
-      possible_reorder(president.shares_of(corporation)).take(num_shares).each { |s| move_share(s, swap_to) }
+      totaling_num(possible_reorder(president.shares_of(corporation)), num_shares).each do |s|
+        move_share(s, swap_to)
+      end
       move_share(presidents_share, president)
     end
 
+    def totaling_num(shares, num_shares)
```

Comment:
totaling_num?

Reply:
It returns the set of shares that total the number of shares needed for the presidency swap (which is not equal if there are non-10% shares). It is not a boolean.

Reply:
can you rename it ? i still don't really get it

Reply:
Had to sleep on it to come up with the relevant name.